package/kedro_viz/data_access/repositories/catalog.py

```diff
@@ -59,17 +64,16 @@ def layers_mapping(self):
         if KEDRO_VERSION.match("<0.19.0"):
             if self._catalog.layers is None:
                 self._layers_mapping = {
-                    self.strip_encoding(dataset_name): None
+                    _strip_transcoding(dataset_name): None
                     for dataset_name in self._catalog._data_sets
                 }
             else:
                 for layer, dataset_names in self._catalog.layers.items():
-                    self._layers_mapping.update(
-                        {
-                            self.strip_encoding(dataset_name): layer
-                            for dataset_name in dataset_names
-                        }
-                    )
+                    for dataset_name in dataset_names:
+                        if TRANSCODING_SEPARATOR in dataset_name:
```

Comment:
Why check for the transcoding separator here and not above (line 67)? In either case the check is not necessary as `_strip_transcoding()` handles dataset names that do not contain the transcoding separator, returning `('dataset_name', '')`